tests/components/onkyo/test_config_flow.py

```diff
@@ -619,23 +623,39 @@ async def test_options_flow(hass: HomeAssistant, config_entry: MockConfigEntry)
     config_entry = create_empty_config_entry()
     await setup_integration(hass, config_entry, receiver_info)
 
+    old_volume_resolution = config_entry.options[OPTION_VOLUME_RESOLUTION]
+
     result = await hass.config_entries.options.async_init(config_entry.entry_id)
-    await hass.async_block_till_done()
 
     result = await hass.config_entries.options.async_configure(
         result["flow_id"],
         user_input={
-            "max_volume": 42,
-            "TV": "television",
+            OPTION_MAX_VOLUME: 42,
+            OPTION_INPUT_SOURCES: [],
+        },
+    )
+
+    assert result["step_id"] == "init"
+    assert result["errors"] == {OPTION_INPUT_SOURCES: "empty_input_source_list"}
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            OPTION_MAX_VOLUME: 42,
+            OPTION_INPUT_SOURCES: ["TV"],
+        },
+    )
+
```

Comment:
Let's also assert the type here, since if this ever breaks it will be more descriptive than something like "unknown flow"